### ISSUE TITLE
Magnificent Mercurial Swift - Issue #47 from Update #1 is still present in ConvexSpell

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-07-blueberry-judging#106]
Magnificent Mercurial Swift

medium

# Issue #47 from Update #1 is still present in ConvexSpell
## Summary

Issue [47](https://github.com/sherlock-audit/2023-04-blueberry-judging/issues/47) from update contest 1 was only fixed for CurveSpell but wasn't fixed for ConvexSpell.

## Vulnerability Detail

See issue [47](https://github.com/sherlock-audit/2023-04-blueberry-judging/issues/47)

## Impact

See issue [47](https://github.com/sherlock-audit/2023-04-blueberry-judging/issues/47)

## Code Snippet

[ConvexSpell.sol#L92-L173](https://github.com/sherlock-audit/2023-07-blueberry/blob/main/blueberry-core/contracts/spell/ConvexSpell.sol#L92-L173)

## Tool used

Manual Review

## Recommendation

See issue [47](https://github.com/sherlock-audit/2023-04-blueberry-judging/issues/47)

**Comment by IAm0x52** [comment]
Escalate

This is a completely different issue than #42. This was the same issue from the previous contest. They dev simply missed the occurrence in convex spell. This is a valid med and needs to be fixed.

**Comment by sherlock-admin2** [comment]
> Escalate
> 
> This is a completely different issue than #42. This was the same issue from the previous contest. They dev simply missed the occurrence in convex spell. This is a valid med and needs to be fixed.

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by Shogoki** [comment]
> Escalate
> 
> This is a completely different issue than #42. This was the same issue from the previous contest. They dev simply missed the occurrence in convex spell. This is a valid med and needs to be fixed.

Agree, not a duplicate #42. This is an issue from the previous contest that was fixed, but is here found at a different place. Can be a valid finding on its own.

**Comment by IAm0x52** [comment]
Missed that this should be a dupe of #60

**Comment by hrishibhat** [comment]
Result:
Medium
Unique

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [IAm0x52](https://github.com/sherlock-audit/2023-07-blueberry-judging/issues/106/#issuecomment-1694745204): accepted

**Comment by hrishibhat** [comment]
Considering this issue a valid issue on its own. #60 talks of the fix that was implemented in the previous contest, while this issue points to the original underlying issue that still existed in one of the contracts. 
This is a unique situation, hence both the issues are treated separately

**Comment by nevillehuang** [comment]
> #1

Hi @hrishibhat both this issue #106 and #60 are talking about the same root cause of lack of approval amounts as the previous issue highlighted [here](https://github.com/sherlock-audit/2023-04-blueberry-judging/issues/47), so they should be duplicated

**Comment by nevillehuang** [comment]
> @nevillehuang #60 want to fix previous fix which is wrong.

Retracting my comment, I see where the distinction is now.

One talks about potentially DoSing approval by front-running and transferring funds, but it seems to be invalid since A.balance(address(this)) is always used.
